### ISSUE TITLE
Selecting multiple buildings

### ISSUE THREAD

**Issue by thany** [openstreetmap/iD#8552]
Cannot select multiple buildings.

Lasso select ALWAYS selects points, and NEVER buildings. Even when EVERYTHING except buildings are hidden using the Map Data panel. It always insists on selecting points.

It also doesn't even select points when zoomed out to some point. It then just becomes a fancy drawing tool that doesn't do literally anything.

**Comment by 1ec5** [comment]
This is a duplicate of #7375.

As of #8264, there’s a shortcut you can press to select the buildings instead of their vertex nodes. Drag a ring around the buildings while holding down Shift, then press <kbd>|</kbd>. This feature hasn’t been released yet, but you can try it out in the [development preview](https://ideditor.netlify.app/) for now.

**Comment by thany** [comment]
Press what? Is that an l, 1 or | ? 🙂

But in that development preview, could someone then also make it so that selection never selects hidden features? I dunno if that's really hard to do or something, but it seems pretty basic to me...

**Comment by 1ec5** [comment]
> Press what? Is that an l, 1 or | ? 🙂

Sorry for the late reply. That was a vertical bar. Thankfully, the shortcut has been changed back to Ctrl plus the up arrow (⌘↑ on macOS), and it is now available on osm.org.

> could someone then also make it so that selection never selects hidden features?

I can’t reproduce this issue in iD v2.23.0. Is it still selecting hidden features for you?

**Comment by thany** [comment]
Yes it is. Just tried it seconds ago:

![image](https://user-images.githubusercontent.com/152227/196550916-a7069ad1-0a73-4500-af9e-55ab4753d671.png)

On the left & on the map, you can see it has managed to select **points**. On the right, you can see I have unticked points, and only ticked the **buildings** checkbox.

Edit:

I just had a thought. On the one hand, it might be totally logical for it to do this. After all, buildings have to consist of points in order to exist on the map. But then why isn't it *also* selecting the buildings itself? Another thing is that if lasso-selection doesn't care about feature visibility, why isn't it also selecting streets, for example?

In other words: *technically* this might seem perfectly sensible. In terms of user experience, one might argue that this doesn't make any sense at all, because generally a user doesn't care about technical specificities 🙂
